app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -20,24 +21,35 @@ class EventRepositoryImpl @Inject constructor(
 ) : EventRepository {
 
     override suspend fun createEvent(event: AgendaItem.Event) {
-        getAuthResult {
+        val result = getAuthResult {
             api.createEvent(
                 eventData = MultipartBody.Part
-                    .create(Json.encodeToString(event.toCreateEventRequest()).toRequestBody()),
-                photoData = event.photos?.map { eventPhoto ->
-                    when (eventPhoto) {
-                        is EventPhoto.Local -> {
-                            MultipartBody.Part
-                                .create(eventPhoto.uri.toFile().readBytes().toRequestBody())
-                        }
-                        is EventPhoto.Remote -> {
-                            MultipartBody.Part
-                                .create(eventPhoto.photoUrl.encodeToByteArray().toRequestBody())
-                        }
-                    }
-                } ?: emptyList()
+                    .createFormData(
+                        name = "create_event_request",
+                        value = Json.encodeToString(event.toCreateEventRequest())
+                    ),
+                photoData = event.photos.filterIsInstance<EventPhoto.Local>().mapIndexed {
+                    index, eventPhoto ->
+                    MultipartBody.Part
+                        .createFormData(
+                            name = "photo$index",
+                            filename = eventPhoto.key,
+                            body = eventPhoto.uri.toFile().readBytes().toRequestBody()
```

Comment:
Have you tried this? I think you can only convert file uris that start with `file://` to a file. If you pick an image from the gallery it's as content uri from which you get the byte array using `contentResolver`